app/models/king.rb

```diff
@@ -1,8 +1,31 @@
 class King < Piece
 
   def is_valid?(new_x, new_y)
-    return false unless super 
+    return false unless super
+    if moved == false && new_y == y_coordinate && (x_coordinate - new_x).abs == 2

```

Comment:
a few suggestions here:
1. `moved == false` can be just `!moved`... 
2. `new_y == y_coordinate && (x_coordinate - new_x).abs == 2` : can we find a name for this condition and extract it out to a separate small method?